README.md

```diff
@@ -33,106 +39,102 @@ input:
             type_id: string
 steps:
   example:
-    plugin: ghcr.io/janosdebugs/arcaflow-example-plugin
-    # step: step-id if the plugin has more than one step
-    # deploy:
-    #   type: docker|kubernetes
-    #   ... more options
+    plugin:
+      deployment_type: image
+      src: quay.io/arcalot/arcaflow-plugin-example
     input:
       name: !expr $.input.name
-output:
-  message: !expr $.steps.example.outputs.success.message
+outputs:
+  success:
+    message: !expr $.steps.example.outputs.success.message
 ```
 
-As you can see, it has a `version`, `input`, a list of `steps`, and an `output` definition. Each of these keys is required in a workflow. These can be linked together using JSONPath expressions (not all features are supported). The expressions also determine the execution order of plugins.
+As you can see, a workflow has the root keys of `version`, `input`, `steps`, and
+`outputs`. Each of these keys is required in a workflow. These can be linked together
+using the Arcaflow
+[expression language](https://arcalot.io/arcaflow/workflows/expressions/). The
+expressions also determine the execution order of plugins.
 
-You can now create an input YAML for this workflow: (save it to input.yaml)
+An input file for this basic workflow may look like:
 
 ```yaml
 name: Arca Lot
 ```
 
-If you have a local Docker / Moby setup installed, you can run it immediately:
+The Arcaflow engine uses a configuration to define the standard behaviors for deploying
+plugins within the workflow. The default configuration will use Docker as the container
+image deployer and will set the log outputs to the `info` level.
 
-```
-./arcaflow -input input.yaml
+If you have a local Docker / Moby setup installed, you can run the workflow immediately:
+
+```bash
+arcaflow -input input.yaml
 ```
 
-If you don't have a local Docker setup, you can also create a `config.yaml` with the following structure:
+If you don't have a local Docker setup, or if you want to use another deployer or any
+custom configuration parameters, you can create a `config.yaml` with your desired
+parameters. For example:
 
 ```yaml
 deployers:
   image: 
-    deployer_name: docker|podman|kubernetes
+    deployer_name: podman
+    deployment:
+      host:
+        NetworkMode: host
+      imagePullPolicy: IfNotPresent
   python:
     deployer_name: python
-  # More deployer options
 log:
-  level: debug|info|warning|error
+  level: debug
+logged_outputs:
+  error:
+    level: debug
 ```
 
 You can load this config by passing the `-config` flag to Arcaflow.
```

Comment:
I'd prefer to use `--config`, as the ability to drop a hyphen is a rather bizarre Go oddity. If they know it's Go `flag` and want to take advantage of that non-standard behavior, that's on them, but I'd prefer not to document it as expected/supported behavior.

Reply:
OK, I kind of agree, and I certainly don't mind documenting with the double-dash notation.